### ISSUE TITLE
InteractiveViewer does not support mouse scrolling

### ISSUE THREAD

**Issue by jezell** [flutter/flutter#121961]
According to the comment closing #64210, InteractiveViewer should support mouse scrolling. @justinmc 

However, the referenced pull request #112171 explicitly checks for trackpad as the source of the scroll and forces scale behavior if the device is a mouse. 

Testing with beta channel, pans fine on my Mac trackpad, but scroll wheel on mouse still scales. Stepped through the code and it's skipping the scroll code due to this line in interactive_viewer.dart: 

```dart
     if (event.kind == PointerDeviceKind.trackpad) {
```

**Comment by exaby73** [comment]
Hello @jezell. Currently this is working as expected. For your use case, see the upcoming feature for 2D scrolling (https://www.youtube.com/watch?v=UDZ0LPQq-n8). Since 2D scrolling is work in progress, and there aren't any plans to update `InteractiveViewer` to support scrolling (see https://github.com/flutter/flutter/issues/119787#issuecomment-1414503611), I will close this issue. If you disagree, please comment with your reasoning, and I can reopen the issue. Thank you

**Comment by jezell** [comment]
@exaby73 I'm not referring to 2D scrolling and this is not working as expected. The issue #64210 was closed saying that mouse scrolling is in fact supposed to be supported by the commit that was merged that adds natural trackpad scrolling, so there seems to be some disagreement here about what is the expected behavior. 

It makes no sense that if I have a mouse attached and a trackpad attached to the same machine the trackpad now properly scrolls the interactive viewer instead of scaling but using the mouse wheel to scroll zooms. It's inconsistent behavior between the two input devices on the same machine with the same widget. Refer to Figma, Miro, Google Drawings for the correct behavior. In all of these apps, the mouse wheel scrolls up and down vertically it does not zoom. That is the expected behavior. The mouse wheel is for used for scrolling and users expect it to scroll.

**Comment by jezell** [comment]
@justinmc can you clarify here what you meant when you said: 

`Now that https://github.com/flutter/flutter/pull/112171 and related PRs are merged, scrolling on a mouse or trackpad causes a pan, and pinching on a trackpad causes a scale. There are several other suggestions in this issue (like the https://github.com/flutter/flutter/issues/64210#issuecomment-1373207786), but I'm going to consider this issue complete.`

**Comment by justinmc** [comment]
I agree that this should be reopened.  We should probably support panning with the mousewheel one way or another.  

That said, @exaby73 is right that real scrolling is a separate thing that InteractiveViewer will likely never support.  It might be better in some use cases to wait for 2D scrolling or to wrap a tall InteractiveViewer in a scrollable widget.

But if you do want InteractiveViewer to pan with the mouse wheel, maybe we could turn [trackpadScrollCausesScale](https://master-api.flutter.dev/flutter/widgets/InteractiveViewer/trackpadScrollCausesScale.html) into an enum or something that configures how you want scrolling to work?  Really I wish users could handle this themselves with GestureDetectors, but that's probably not feasible right now without a big refactor.

I think the behavior we were mirroring was Google Maps, which does zoom with the mouse wheel.  I can't remember what it does with a trackpad though.

**Comment by jezell** [comment]
@justinmc yes, I think trackpadScrollCausesScale should just be consistent here between trackpad and mouse wheel. I agree 2D scrollbars are a totally different issue, but it's really weird that the setting doesn't result in consistent behavior between mouse and trackpad, since it's really the same problem being addressed that users expect scrolling gestures and scroll wheels to pan, not zoom. If there's some case where it makes sense for the behavior to not be consistent, mouseWheelScrollCausesScale could be introduced.

**Comment by exaby73** [comment]
@jezell Thanks for the clarification here. @justinmc If I am not mistaken, can this issue be a considered a proposal to streamline the scrolling API on InteractiveViewer? Or a proposal to make `trackpadScrollCausesScale` affect mouse scroll wheel? Or both?

**Comment by justinmc** [comment]
Yes to both!

**Comment by jezell** [comment]
Sounds good!

**Comment by andykim78** [comment]
I've found a bug where Interactive Viewer doesn't scroll, but rather zooms when scrolling with my Macbook Air's trackpad when running a web build of a Flutter app. The weird thing is that it only happens when the browser window is on the built-in display and it scrolls properly when the window is on an external display.

I tracked down what the cause is and when using the built-in trackpad or Magic Mouse with the window on the built-in display, the scroll event device comes in as PointerDeviceKind.mouse instead of PointerDeviceKind.trackpad but a PointerScrollEvent event is still routed to _InteractiveViewerState._receivedPointerSignal method, not matching the `if (event.kind == PointerDeviceKind.trackpad && !widget.trackpadScrollCausesScale) {` condition, thus causing the scale.

It seems that either mouse scrolling to pan should be supported sooner rather than later to fix this, or the event should get properly routed as a PanAndZoom event.

OS: macOS Sonoma (14.0)
Google Chrome: 118.0.5993.88

**Comment by espresso3389** [comment]
I'm developing [a PDF viewer plugin](https://github.com/espresso3389/pdfrx) using InteractiveViewer and I need mouse-wheel to scroll support on it.
I think TwoDimensionalScrollable is not suitable for document viewer that needs pinch-zooming. Matrix based view transforms works well for my purpose.

Currently, I've forked the original InteractiveViewer but if InteractiveViewer supports some kind of customization on mouse-wheel, it's very helpful for me.

**Comment by ffpresents** [comment]
I have faced the exact same problem.

@espresso3389, your code exactly solved my problem. Awesome! Thank you very much.
